src/tool_writeout_json.c

```diff
@@ -103,13 +103,20 @@ static int writeString(FILE *str, CURL *curl, const char *key, CURLINFO ci)
   return 0;
 }
 
-static int writeLong(FILE *str, CURL *curl, const char *key, CURLINFO ci)
+static int writeLong(FILE *str, CURL *curl, const char *key, CURLINFO ci,
+        struct per_transfer *per)
 {
-  long val = 0;
-  if(CURLE_OK == curl_easy_getinfo(curl, ci, &val)) {
-    fprintf(str, "\"%s\":%ld", key, val);
+  if(!ci && !strcmp(key, "num_headers")) {
```

Comment:
If you change this function prototype to get a `struct writeoutvar *e` argument instead of `char *key` you could compare with `e->id == VAR_NUM_HEADERS` instead of a strcmp(), which seems like a nicer approach that doesn't distribute the string in multiple places in the code.